tests/test_mongo_provider.py

```diff
@@ -0,0 +1,219 @@
+# =================================================================
+#
+# Authors: Timo Tuunanen <timo.tuunanen@rdvelho.com>
+#
+# Copyright (c) 2019 Timo Tuunanen
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+#
+# =================================================================
+
+import pytest
+
+from pygeoapi.provider.mongo import MongoProvider
+
+monogourl = 'mongodb://localhost:27017/'
+mongodb = 'testdb'
+mongocollection = 'testplaces'
+
+
+@pytest.fixture()
+def config():
+    return {
+        'name': 'Mongo',
+        'data': {
```

Comment:
1. For consistency (see Elasticsearch provider), can `data` be in the Connection String URI format (see https://docs.mongodb.com/manual/reference/connection-string/) ?
2. Can the `fields` property be populated dynamically (instead of hard coding in config)?

Reply:
1. Updated as requested.

2. Updated get_fields to be dynamic.